src/nginxconfig/generators/conf/drupal.conf.js

```diff
@@ -18,8 +18,10 @@ export default global => {
     const config = {};
 
     config['# Drupal: deny private files'] = '';
-    config['location ~ ^/sites/.*/private/'] = {
+
+    config['location ~ ((^|/)\.|^.*\.yml$|^/sites/.*/private/|^/sites/[^/]+/.*settings.*\.php$)'] = {
```

Comment:
In `^/sites/[^/]+/.*settings.*\.php$`, `[^/]+` is used which I assume is meant to handle a single-level-deep subdirectory. This on its own would be fine, however after is `/.*` which will match anything including multi-level-deep subdirectories.

```suggestion
    config['location ~ ((^|/)\.|^.*\.yml$|^/sites/.*/private/|^/sites/[^/]+/[^/]*settings.*\.php$)'] = {
```

Assuming the intention is a single-level-deep subdirectory here, this should fix it.